### ISSUE TITLE
Firebase analytics symbols not loading in IOS proj

### ISSUE THREAD

**Issue by jfversluis** [xamarin/xamarin-macios#21817]
---
**Issue moved from dotnet/maui#26660**
- Please respond to @sasupallisantosh.

---

_From @sasupallisantosh on Tuesday, December 17, 2024 2:34:11 AM_

Previously, I encountered a build issue with the MAUI iOS application due to the plugin.firebase (https://github.com/dotnet/maui/issues/26455) After resolving the issue, the build was successful. However, when I connected to a Mac and ran the iOS application on a simulator, I received the following error:
`clang++ exited with code 1: Undefined symbols for architecture arm64: "_APMAnalyticsConfiguration", referenced from: +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMAppMeasurementOriginFirebase", referenced from: +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMConsentSettings3P", referenced from: +[FIRAnalytics setConsent:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMFormattedEventName", referenced from: +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMFormattedUserPropertyName", referenced from: +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMIsAnalyticsCollectionDeactivated", referenced from: +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMIsAnalyticsCollectionEnabled", referenced from: +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMMonitorLogTagOptionKey", referenced from: +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMUserDataFieldEmailAddress", referenced from: +[FIRAnalytics initiateOnDeviceConversionMeasurementWithEmailAddress:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMUserDataFieldHashedEmailAddress", referenced from: +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedEmailAddress:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMUserDataFieldHashedPhoneNumber", referenced from: +[FIRAnalytics initiateOnDeviceConversionMeasurementWithHashedPhoneNumber:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o) "_APMUserDataFieldPhoneNumber", referenced from: +[FIRAnalytics initiateOnDeviceConversionMeasurementWithPhoneNumber:] in FirebaseAnalytics[arm64][10](FIRAnalytics.o)` 

Please help me resolve this issue.

**Comment by rolfbjarne** [comment]
Please get an [MSBuild binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) that shows the build error, and we can start diagnosing what's going wrong.

**Comment by sasupallisantosh** [comment]
Hi @rolfbjarne ,
Without a Mac connection, the build was successful. However, when we connected a Mac and tried to build or run, we encountered an error. I tried building it using the command line, and it showed the build as successful. I will attach the MSBuild file and paste the link below for my sample project, which reproduces the same error.

[msbuild.zip](https://github.com/user-attachments/files/18191277/msbuild.zip)

Git Pub Repo: https://github.com/sasupallisantosh/PushTest.git

**Comment by rolfbjarne** [comment]
> However, when we connected a Mac and tried to build or run, we encountered an error.

Can you get an [MSBuild binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#visual-studio-for-windows) for this failure as well? The link explains how to get a binlog for a build from within Visual Studio.

**Comment by sasupallisantosh** [comment]
I have mentioned in previous comment msbuild info file,
https://github.com/user-attachments/files/18191277/msbuild.zip

**Comment by rolfbjarne** [comment]
> I have mentioned in previous comment msbuild info file, [user-attachments/files/18191277/msbuild.zip](https://github.com/user-attachments/files/18191277/msbuild.zip)

That binlog shows a successful build, not a failing build.

**Comment by sasupallisantosh** [comment]
Yes, @rolfbjarne , the build was successful. I tried building it using the command line, and it showed the build as successful and generated the MSBuild file successfully. However, when we connected a Mac and tried to run it, the error occurred.

**Comment by rolfbjarne** [comment]
> Yes, @rolfbjarne , the build was successful. I tried building it using the command line, and it showed the build as successful and generated the MSBuild file successfully. However, when we connected a Mac and tried to run it, the error occurred.

If you're referring to this error: https://github.com/dotnet/maui/issues/26660, then that's a build error, and it will show up in the binlogs created by Visual Studio (if you enable binlog creation in Visual Studio as explained here: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#visual-studio-for-windows).

**Comment by sasupallisantosh** [comment]
Yes I enabled the bin log , when we will connect the mac we got  build
issue without mac connection build was successful, when we run the
application from command terminal build was successful, when we will run
the project on iOS simulator we got that error.

**Comment by rolfbjarne** [comment]
Can you zip up and attach all the binlogs created when building and running using Visual Studio?

**Comment by chrmeehansr** [comment]
With all the references to different github threads, I'm not sure where to ask if someone got this resolved. I'm encountering the same challenge. dotnet build on Windows device, no errors.. deploy to mac paired and get the same error. Hoping someone has a solution and or thread with latest I can read. Thank you. 

-C

**Comment by TimothyFran** [comment]
Having the same problem

**Comment by bpaquin** [comment]
same

**Comment by zer0Ambit1on** [comment]
same

**Comment by Quaybe** [comment]
Why are all of this issues closed when it seems like no one has a solution?

**Comment by rolfbjarne** [comment]
For anybody who runs into this: please get an [MSBuild binlog](https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#binary-build-logs) that shows the build error, and we can start diagnosing what's going wrong.

**Comment by flipper09112** [comment]
+1
